scenario/scripts/snapshot.py

```diff
@@ -0,0 +1,752 @@
+        logger.debug(
+            f"  getting relation data for endpoint {raw_relation.get('endpoint')!r}"
+        )
+        related_units = raw_relation["related-units"]
+        #    related-units:
+        #      owner/0:
+        #        in-scope: true
+        #        data:
+        #          egress-subnets: 10.152.183.130/32
+        #          ingress-address: 10.152.183.130
+        #          private-address: 10.152.183.130
+
+        relation_id = raw_relation["relation-id"]
+
+        local_unit_data_raw = _juju_exec(
+            target, model, f"relation-get -r {relation_id} - {target} --format json"
+        )
+        local_unit_data = json.loads(local_unit_data_raw)
+        local_app_data_raw = _juju_exec(
+            target,
+            model,
+            f"relation-get -r {relation_id} - {target} --format json --app",
+        )
+        local_app_data = json.loads(local_app_data_raw)
+
+        some_remote_unit_id = Target(next(iter(related_units)))
+        relations.append(
+            Relation(
+                endpoint=raw_relation["endpoint"],
+                interface=_get_interface_from_metadata(
+                    raw_relation["endpoint"], metadata
+                ),
+                relation_id=relation_id,
+                remote_app_data=raw_relation["application-data"],
+                remote_app_name=some_remote_unit_id.app_name,
+                remote_units_data={
+                    Target(tgt).unit_id: _clean(val["data"])
+                    for tgt, val in related_units.items()
+                },
+                local_app_data=local_app_data,
+                local_unit_data=_clean(local_unit_data),
+            )
+        )
+    return relations
+
+
+def get_model(name: str = None) -> Model:
+    logger.info("getting model...")
+
+    jsn = _juju_run("models")
+    model_name = name or jsn["current-model"]
+    try:
+        model_info = next(
+            filter(lambda m: m["short-name"] == model_name, jsn["models"])
+        )
+    except StopIteration as e:
+        raise InvalidModel(name) from e
+
+    model_uuid = model_info["model-uuid"]
+    model_type = model_info["type"]
+
+    return Model(name=model_name, uuid=model_uuid, type=model_type)
+
+
+def try_guess_charm_type_name():
+    try:
+        charm_path = Path(os.getcwd()) / "src" / "charm.py"
+        if charm_path.exists():
+            source = charm_path.read_text()
+            charms = re.compile(r"class (\D+)\(CharmBase\):").findall(source)
+            if len(charms) < 1:
+                raise RuntimeError(f"Not enough charms at {charm_path}.")
+            elif len(charms) > 1:
+                raise RuntimeError(f"Too many charms at {charm_path}.")
+            return charms[0]
+    except Exception as e:
+        logger.warning(f"unable to guess charm type: {e}")
+    return None
+
+
+class FormatOption(str, Enum):
+    state = "state"
+    json = "json"
+    pytest = "pytest"
```

Comment:
Why is this enum inheriting from both `str` and `Enum`? Looking through the code, it seems fine to just inherit from `Enum`. Is this something that has to do with the typer library?